src/views/Export.vue

```diff
@@ -1,92 +1,168 @@
-    </div>
-    <boxed-button :disabled="downloading" v-on:click="download" />
-    <div id="error">{{ error }}</div>
-    <p :hidden="!downloading">Download is in progress...</p>
-    <action-button-group
-      id="done"
-      :require-confirmation="true"
-      v-on:submitted="handler()"
-      action-button-text="DONE"
-    />
-  </app-card>
+  <v-container fluid>
+    <v-col cols="12" class="px-8 mx-auto export-form">
+      <v-row>
+        <v-col class="pl-0">
+          <v-btn rounded text @click="goHome()">
+            <v-icon left>arrow_back</v-icon>
+            Back
+          </v-btn>
+        </v-col>
+      </v-row>
+
+      <v-row>
+        <v-col class="pb-0">
+          <h1>Export</h1>
+          <p>
+            Please select a date range to export the data.
+          </p>
+        </v-col>
+      </v-row>
+
+      <v-row>
+        <v-form>
+          <v-col>
+            <date-quick-select
+              class="pb-3"
+              @select="quickSelect($event)"
+              :edited="edited"
+            />
+            <date-picker-menu
+              v-model="startDate"
+              label="Start date"
+              prepend-icon="today"
+              @input="edited = true"
+            />
+            <date-picker-menu
+              v-model="endDate"
+              label="End date"
+              prepend-icon="event"
+              @input="edited = true"
+            />
+          </v-col>
+        </v-form>
+      </v-row>
+
+      <v-row>
+        <v-col>
+          <p>Click "Download" to export data as an Excel spreadsheet.</p>
+          <v-btn color="primary" @click="download" :loading="downloading">
+            <v-icon left>file_download</v-icon>Download
+          </v-btn>
+        </v-col>
+      </v-row>
+      <div id="error">{{ error }}</div>
+      <p :hidden="!downloading">Download is in progress...</p>
+    </v-col>
+  </v-container>
 </template>
 
 <script lang="ts">
-import Component from "vue-class-component";
-import AppCard from "@/components/AppCard.vue";
-import ActionButtonGroup from "@/components/ActionButtonGroup.vue";
-import BoxedButton from "@/components/BoxedButton.vue";
 import Vue from "vue";
+import Component from "vue-class-component";
 import * as firebase from "firebase/app";
 import "firebase/functions";
 import "firebase/storage";
 import moment from "moment";
+// components
+import DatePickerMenu from "@/components/Export/DatePickerMenu.vue";
+import DateQuickSelect from "@/components/Export/DateQuickSelect.vue";
 
 @Component({
   components: {
-    ActionButtonGroup,
-    AppCard,
-    BoxedButton,
+    DatePickerMenu,
+    DateQuickSelect,
   },
 })
 export default class Settings extends Vue {
-  active = false;
   downloading = false;
-  end_date = moment().format("YYYY-MM-DD");
-  start_date = moment().subtract(6, "days").format("YYYY-MM-DD");
+
+  startDate = moment().startOf("week").format("YYYY-MM-DD");
+
+  endDate = moment().format("YYYY-MM-DD");
```

Comment:
maybe `startDate` and `endDate` should be moments that get formatted later? there's a lot of repeated `.format("YYYY-MM-DD")` code that then requires extra constants (like line 107) and string comparisons (like line 144)

Reply:
hmm maybe, but I need it as a `string` since I use it as a `v-model` for the date pickers. I could simplify the amount of times I call format within the `quickSelect()` function though by doing something like this?

```typescript
start = moment()
end = moment()
if( ... ) {
  start = ... 
  end = ...
}
...
this.startDate = start.format("YYYY-MM-DD")
this.endDate = end.format("YYYY-MM-DD")
```